### ISSUE TITLE
Label or Place holder value is not given for edit fields present on New recipe page: A11y_.NET Core_RecipesAppHomeScreen_New Recipe_AI4MAC

### ISSUE THREAD

**Issue by kapilvaishna** [dotnet/maui#22906]
**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**

"[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier."
### GitHub Tags
#A11yTCS;#A11yMAS;#WCAG4.1.2;#AI4Mac;#E2E_.NETCore_Mac_Jun2024;#.NET Core;#Name Role Value;#DesktopApp;#Device;#FTP;#A11ySev2;#Mac;#Reopened;#Regressed:10-09-24; 

## Environment Details:
Application: .NET Core
Version 2 (2)
Operating System: Sonoma 14.5 

## Repro Steps:

1. Install and open ".NET MAUI" app.
2. TAB to "My Recipes" tab and Press ENTER key.
3. TAB to "New" button and press ENTER key.
4. "New Recipe" Screen will open.
5. TAB through edit fields and inspect them.
6. Observe that Label or Place holder value is not given for edit fields present on New recipe page.

## Actual Result:
Label or Place holder value is not given for edit fields present on New recipe page.
### Similar issue observed in following scenario:

1. Recipes App My Recipes Edit_Edit Recipe

## Expected Result:
Label or Place holder value should be assign to edit fields present on New recipe page.
## User Impact:
Screen reader user will face issue and speech input user will not be able to interact with control. 
### Attachment

https://github.com/dotnet/maui/assets/124254981/d9eefc45-af9e-4d88-8dc9-0d04a7185d4e

Similar issue

https://github.com/dotnet/maui/assets/124254981/b5b3a6ed-83d0-4b8e-bf66-62609d3e0da0

**Comment by kevinxufei** [comment]
Verified this issue on Visual Studio for Mac 17.6.12 build 410 (8.0.40 & 8.0.6). Can repro it.

**Comment by kupatkar99** [comment]
GithubTags:#Rev:yaja;

**Comment by Yash14j** [comment]
GithubTags:#Rev:yaja;

**Comment by tj-devel709** [comment]
I see that using the Semantics.Description will add the "Label" property in the Accessibility Inspector but I am not too sure if the value of a "Placeholder Value" makes sense inside a UITextView (Maui Editor) since there is no Placeholder property on UITextView in the first place. I am trying to find a mac app that has a UITextView with this Placeholder property and am not finding one. Do you, @kapilvaishna have an example of this?

**Comment by tj-devel709** [comment]
On the AppStore app, the place where you leave a review uses a UITextField and not a UITextView.

Even this textbox I type in uses a UITextView but has the placeholder value as empty:
![image](https://github.com/user-attachments/assets/cc23b2f0-93df-472e-8424-c10e0d36bfb9)

So an example if you can find one would be very helpful in solving this issue :)

**Comment by kapilvaishna** [comment]
@tj-devel709 just add the label only for all controls and resolve it i will verify and close this bug as i mentioned only one thing "label or placeholder" is required.  and i think it will not reproduce any other issue for screen reader user or speech input user if label is only their. and if it makes any other issue for screen reader user or speech input user than first i will find an example and then i will log new bug.

**Comment by tj-devel709** [comment]
@kapilvaishna it looks like for the Maui Entries, they cannot have the Placeholder and the SemanticProperties.Description as the voice over will read both on iOS devices. If they either need the 'label' or 'placeholder' values, would it be fine if those just had the placeholder value as well?

**Comment by kapilvaishna** [comment]
@tj-devel709 if you will provide just placeholder value then i think screen reader user will not get any issue but if label is not there than speech input user will get issue, they will not be able to interact with that particular control.

I can see label is given for review text filed what is the issue if you do same for other text filed also.

**Comment by tj-devel709** [comment]
Ah okay, I am testing out the Voice Control on mac now and see what you mean. When I don't add the SemanticProperties.Description on the Entry, I cannot tell the Voice Control to 'click Image URL' if it just has it as the placeholder value..

I think the difference between the Review field and the Image URL field is that they are mapped to two different iOS controls that behave differently. The Review field is a UITextView which does not actually have a placeholder property, but we add a MAUI property there to use in place of it so when iOS is reading the field, it just sees the SemanticProperties.Description. 
The Image URL field is a UITextField that does actually have an iOS placeholder value so when the voice over passes it on iOS, it reads the Placeholder and SemanticProperties.Description.

**Comment by kapilvaishna** [comment]
@tj-devel709 You can try with AIRIA label to provide the label for "Recipe name" and "Image URL" edit filed.
You need to provide label for "Recipe name" and "Image URL" edit filed to close this bug.
Verified the issue in version 3

https://github.com/user-attachments/assets/2b23e008-b78f-43ad-a43a-a6d08f4ce162

**Comment by tj-devel709** [comment]
Hi @kapilvaishna, can you confirm the version you are testing? I am seeing this on my end:

https://github.com/user-attachments/assets/776f9a51-879d-4872-977b-37dc69451d7e

**Comment by kapilvaishna** [comment]
Issue is repro in version 3(3)
<img width="1680" alt="Screenshot 2024-10-03 at 5 39 44 PM" src="https://github.com/user-attachments/assets/cdfa78ce-7938-4cbd-bc33-cab3c6ff7153">

**Comment by kapilvaishna** [comment]
@tj-devel709 I can see now it is fixed for you can you please let me in which version you are verifying. it is reproducing for me in version 3(3).

**Comment by PureWeen** [comment]
@kapilvaishna I just submitted a version 4 that should be available once it's approved in test flight

**Comment by kapilvaishna** [comment]
Issue is not repro in version 4(4) so closing this bug.
https://github.com/user-attachments/assets/b4253b3f-8a83-4a69-8671-4df1011b71bc
